src/irmin/tree_intf.ml

```diff
@@ -256,8 +256,9 @@ module type S = sig
       collection of marks [m] to store the cache of keys: the fold will modify
       [m]. This can be used for incremental folds. *)
 
-  type 'a node_fn = path -> step list -> 'a -> 'a Lwt.t
-  (** The type for {!fold}'s [pre] and [post] parameters. *)
+  type ('a, 'b) folder = path -> 'b -> 'a -> 'a Lwt.t
```

Comment:
Typically documentation only updates don't need a changelog entry, but in this case it might be worth adding an entry to mention that the `node_fn` type has been replaced with `folder` and that `folder` is now `cps_folder`

Reply:
Good point. I'll add an entry about the replacement of `node_fn` by a more generic `folder` type. What is now `cps_folder` is strictly an internal type to `tree.ml`, so I'll leave that out of the changelog.